src/TAction.cpp

```diff
@@ -121,7 +121,7 @@ bool TAction::setScript(const QString& script)
 bool TAction::compileScript()
 {
     mFuncName = QString("Action") + QString::number(mID);
-    const QString code = QString("function ") + mFuncName + QString("()\n") + mScript + QString("\nend\n");
+    const QString code = QString("function %1() %2\nend").arg(mFuncName, mScript);
```

Comment:
```suggestion
    const QString code = qsl("function %1() %2 end").arg(mFuncName, mScript);
```

Reply:
Initially I did this same thing and then found that the last line of the edited box having a comment would comment out the end.  I don't like how an unclosed thing would say error found on line 11 if there are 10 lines in the edit box, but this may be a necessary evil.

Reply:
Oh, yeah, I see what you mean - so that isn't going to work. *\*sigh\**

Reply:
I did not notice that when I took the screenshot.  You are exactly right.  I think I did not touch that in the current html escaping PR because I failed to produce such an error message in that spot.  I put a random `<` at the end of some code and it said `unexpected symbol` without specifying the symbol, and eventually I gave up at trying to produce an error message with a `<` in it.  Thanks very much.  I'll add those to it.